provider/cmd/run.go

```diff
@@ -79,15 +87,113 @@ func RunCmd() *cobra.Command {
 		return nil
 	}
 
+	cmd.Flags().String(FlagBidPricingStrategy, "scale", "Pricing strategy to use")
+	if err := viper.BindPFlag(FlagBidPricingStrategy, cmd.Flags().Lookup(FlagBidPricingStrategy)); err != nil {
+		return nil
+	}
+
+	cmd.Flags().Uint64(FlagBidPriceCPUScale, 0, "cpu pricing scale in uakt")
+	if err := viper.BindPFlag(FlagBidPriceCPUScale, cmd.Flags().Lookup(FlagBidPriceCPUScale)); err != nil {
+		return nil
+	}
+	cmd.Flags().Uint64(FlagBidPriceMemoryScale, 0, "memory pricing scale in uakt")
+	if err := viper.BindPFlag(FlagBidPriceMemoryScale, cmd.Flags().Lookup(FlagBidPriceMemoryScale)); err != nil {
+		return nil
+	}
+	cmd.Flags().Uint64(FlagBidPriceStorageScale, 0, "storage pricing scale in uakt")
+	if err := viper.BindPFlag(FlagBidPriceStorageScale, cmd.Flags().Lookup(FlagBidPriceStorageScale)); err != nil {
+		return nil
+	}
+	cmd.Flags().String(FlagBidPriceScriptPath, "", "path to script to run for computing bid price")
+	if err := viper.BindPFlag(FlagBidPriceScriptPath, cmd.Flags().Lookup(FlagBidPriceScriptPath)); err != nil {
+		return nil
+	}
+	cmd.Flags().Uint(FlagBidPriceScriptProcessLimit, 32, "limit to the number of scripts run concurrently for bid pricing")
+	if err := viper.BindPFlag(FlagBidPriceScriptProcessLimit, cmd.Flags().Lookup(FlagBidPriceScriptProcessLimit)); err != nil {
+		return nil
+	}
+	cmd.Flags().Uint(FlagBidPriceScriptTimeout, 10000, "exedcution timelimit for bid pricing script in milliseconds")
+	if err := viper.BindPFlag(FlagBidPriceScriptTimeout, cmd.Flags().Lookup(FlagBidPriceScriptTimeout)); err != nil {
+		return nil
+	}
+
 	return cmd
 }
 
+const (
+	bidPricingStrategyScale       = "scale"
+	bidPricingStrategyRandomRange = "randomRange"
+	bidPricingStrategyShellScript = "shellScript"
+)
+
+var allowedBidPricingStrategies = [...]string{
```

Comment:
Making them types eg: `type bidPricingStrategy string`  then declare them as `var scale bidPricingStrategy = "scale"`; it would make the comparison between types, and not raw string data. 

Then it becomes trivial to use a type switch for selection, and easy default clause for the errNoSuchBidPricing...

Unless there's some issue with the viper flags that this pattern escapes with :grimacing:

Reply:
Are you suggesting I'd cast from a string into `bidPricingStrategy` ?

Reply:
That's what I was effectively saying, but realized since we're just representing a state, we don't really need the string aspect to describe if we use a stringification. For example
```
//go:generate stringer -linecomment -output=autogen_stringer.go -type=bidPricingStrategy

type bidPricingStrategy uint8

const (
	scaling bidPricingStrategy = iota + 1
	randomRange
	shellsScript
)
```

Then by running `go generate ...` we'll be able an access a string of these types with the generated code:
```go
func _() {
	// An "invalid array index" compiler error signifies that the constant values have changed.
	// Re-run the stringer command to generate them again.
	var x [1]struct{}
	_ = x[scaling-1]
	_ = x[randomRange-2]
	_ = x[shellsScript-3]
}

const _bidPricingStrategy_name = "scalingrandomRangeshellsScript"

var _bidPricingStrategy_index = [...]uint8{0, 7, 18, 30}

func (i bidPricingStrategy) String() string {
	i -= 1
	if i >= bidPricingStrategy(len(_bidPricingStrategy_index)-1) {
		return "bidPricingStrategy(" + strconv.FormatInt(int64(i+1), 10) + ")"
	}
	return _bidPricingStrategy_name[_bidPricingStrategy_index[i]:_bidPricingStrategy_index[i+1]]
}
```

It very nice! ![](https://emojis.slackmojis.com/emojis/images/1510742343/3197/borat.gif?1510742343) Then all the comparisons are between static types, no string comparison necessary, but we get efficient String() formatting for debugging! Not necessary to change, but we've used it in the past.

Reply:
When this becomes more involved we can do a more typical registry pattern.  for now I think it's just fine.